services/sso/src/lib/sso.js

```diff
@@ -19,21 +19,18 @@ const getAzureConfig = (authConfig, logger) => ({
 });
 
 /* istanbul ignore next */
-const loginViaMicrosoft = (req, res, cca, config, logger) => {
+// eslint-disable-next-line consistent-return
+const loginViaMicrosoft = async (req, res, cca, config, logger) => {
   const authCodeUrlParameters = {
     scopes: ["user.read"],
     redirectUri: config.redirectUri,
   };
 
-  cca
-    .getAuthCodeUrl(authCodeUrlParameters)
-    .then((response) => {
-      req.session.isAuthenticated = true;
-      res.redirect(response);
-    })
-    .catch((error) =>
-      logger.error({ error }, `Error getting authorisation url`)
-    );
+  try {
+    return await cca.getAuthCodeUrl(authCodeUrlParameters);
```

Comment:
As the result of `cca.getAuthCodeUrl` is being returned directly we don't need to use `async/await` in this method

Reply:
Removed.